apps/io-wallet-user-func/src/email-provider.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  getMailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import * as t from "io-ts";
+
+import { MailConfig } from "./app/config";
+
+export const SendEmailParams = t.type({
+  html: t.string,
+  subject: t.string,
+  text: t.string,
+  to: t.string,
+});
+
+export type SendEmailParams = t.TypeOf<typeof SendEmailParams>;
+
+export const sendEmail: (
+  params: SendEmailParams,
+) => RTE.ReaderTaskEither<{ mail: MailConfig }, Error, void> =
+  ({ html, subject, text, to }) =>
+  (configs) =>
+    configs.mail.mailFeatureFlag
+      ? pipe(
+          TE.tryCatch(
+            async () => {
+              await sendMail(
```

Comment:
Why are you doing a `trycatch` on `sendMail`?

Reply:
You're right, I think it doesn't make much sense since the error if I'm not mistaken is caught inside the CreateWalletInstanceHandler handler!